vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/common.go

```diff
@@ -353,32 +353,52 @@ func isUnitExists(err error) bool {
 	return isDbusError(err, "org.freedesktop.systemd1.UnitExists")
 }
 
-func startUnit(cm *dbusConnManager, unitName string, properties []systemdDbus.Property) error {
+func startUnit(cm *dbusConnManager, unitName string, properties []systemdDbus.Property, ignoreExist bool) error {
 	statusChan := make(chan string, 1)
+	retry := true
+
+retry:
 	err := cm.retryOnDisconnect(func(c *systemdDbus.Conn) error {
 		_, err := c.StartTransientUnitContext(context.TODO(), unitName, "replace", properties, statusChan)
 		return err
 	})
-	if err == nil {
-		timeout := time.NewTimer(30 * time.Second)
-		defer timeout.Stop()
-
-		select {
-		case s := <-statusChan:
-			close(statusChan)
-			// Please refer to https://pkg.go.dev/github.com/coreos/go-systemd/v22/dbus#Conn.StartUnit
-			if s != "done" {
-				resetFailedUnit(cm, unitName)
-				return fmt.Errorf("error creating systemd unit `%s`: got `%s`", unitName, s)
-			}
-		case <-timeout.C:
+	if err != nil {
+		if !isUnitExists(err) {
+			return err
+		}
+		if ignoreExist {
+			// TODO: remove this hack.
+			// This is kubelet making sure a slice exists (see
+			// https://github.com/opencontainers/runc/pull/1124).
+			return nil
```

Comment:
is there an issue tracking this TODO? the linked issue is from 2016 and is closed... it's not clear what remaining work is needed in runc or in the kubelet

Reply:
@kolyshkin Did we have to update any kubelet calls to include the boolean here?

Reply:
There is no issue created; I am intending to look at it but hasn't yet. For now, nothing needs to be changed, as the old behavior is preserved for kubelet.

TL;DR: systemd driver was modified in https://github.com/opencontainers/runc/pull/1124 to ignore UnitExists error, which is totally wrong (see https://github.com/opencontainers/runc/issues/3780), but since kubelet relies on this behavior, runc 1.1.6 keeps it.

The TODO item needs to be addressed in kubelet first, when runc. There's no rush to do so.